proposals/bound-callable-references.md

```diff
@@ -84,6 +124,42 @@ fun test(c: C?) {
 }
 ```
 
+#### Nullable references
+
+According to this proposal, in the following syntax
+```
+Foo?::bar
+```
+the left-hand side can only be interpreted as a type because `Foo?` is not a valid expression. Therefore this would necessarily be a reference to an extension function named `bar` to the type `Foo?`.
```

Comment:
This `?` business looks scary: what if we introduce such expression at some point (such proposals pop up every now and them)? Can we do something future-proof here?